### PR TITLE
Fix AWS Setup.

### PR DESCRIPTION
**Pull request by brendandburns**
@justinsb @roberthbailey @zmerlynn

### CODE REVIEW

**Comment by roberthbailey** [comment]
Do you need to create one for the kube-proxy as well like we do on GCE (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/configure-vm.sh#L395)?

**Comment by brendandburns** [comment]
It was already creating it for kube-proxy.

(fwiw, I just tested this and it's not quite right, please don't merge yet)

File: cluster/aws/templates/create-dynamic-salt-files.sh [hunk 1]
```diff
@@ -82,6 +82,27 @@ contexts:
...
+apiVersion: v1
+kind: Config
+users:
+- name: kubelet
+  user:
+    token: ${kubelet_token}}
+clusters:
+- name: local
+  cluster:
+     insecure-skip-tls-verify: true
```
Comment (lavalamp) [inline cluster/aws/templates/create-dynamic-salt-files.sh]: is it not possible to do it the right way?

**Comment by roberthbailey** [comment]
> It was already creating it for kube-proxy.

That's what I get for not reading below the fold.

File: cluster/aws/templates/create-dynamic-salt-files.sh (hunk 1 above)
Comment (roberthbailey) [inline cluster/aws/templates/create-dynamic-salt-files.sh]: I think that in AWS the certs are still generated on the master, which means that the kubelet doesn't have the CA cert when it starts up.

**Comment by brendandburns** [comment]
Ok, there was a stray '}' this is now ready for merge.

**Comment by brendandburns** [comment]
I added onto this some fixes that make AWS work again.

**Comment by brendandburns** [comment]
Closes https://github.com/GoogleCloudPlatform/kubernetes/issues/10908

**Comment by justinsb** [comment]
Don't have a lot of context of the kubeconfig commit, but the Salt work LGTM.

**Comment by brendandburns** [comment]
hold on this...

**Comment by justinsb** [comment]
Oh yeah, the "gang of 3" depends on the docker service you've deleted in salt.

Go3 = kube-controller-manager, kube-scheduler, kube-apiserver

**Comment by brendandburns** [comment]
Ok, I re-added the docker service, but set it to enabled: False on aws only.

I don't really know *why* it is causing things to hang (salt is hanging on attempting to start the docker daemon, and that call is a zombie, which causes salt to perma-hang) but this fixes it, my clusters start, since Monit eventually kicks in and restarts Docker.

**Comment by justinsb** [comment]
Afk but I remember docker failing if cbr0 was not yet set up.

**Comment by brendandburns** [comment]
Yeah, docker will fail, but it should hang (I wouldn't think)

But yes, I agree, I think it has something to do with that.

**Comment by brendandburns** [comment]
Ok, this is validated working.  Ready to merge when it looks ok to everyone.

Thanks
--brendan

**Comment by justinsb** [comment]
Salt work LGTM again.  I hope we get #10488 in (systemd support) and we end up not needing this.  Bizarrely, on systemd it works if we do restart the service, although Salt can't do that reliably so we end up relying on an external script.

(I think kubelet actually ends up running/restarting docker after it configures cbr0 - it might not be monit restarting docker.)

**Comment by brendandburns** [comment]
Added LGTM on @justinsb's LGTM, self-adding ok-to-merge, though I wouldn't mind a double check from @dchen1107

**Comment by dchen1107** [comment]
LGTM from my side. Please note that I only make sure there is no breakage for GCE, since I have no idea on AWS configurations. Justine is the guru here. :-)

**Comment by alex-mohr** [comment]
@k8s-bot ok to test

**Comment by alex-mohr** [comment]
@vmarmol ok to merge and tests passed.

@zmerlynn 0.21.1 candidate.

**Pull request merged**